### ISSUE TITLE
Question about the training code

### ISSUE THREAD

**Issue by Epiphqny** [kohjingyu/gill#11]
Hi @kohjingyu , thanks for your code. Can you explain the code in 
https://github.com/kohjingyu/gill/blob/232eb02812e7e138cd7574772a11d3d8a71d34e9/gill/models.py#L312, 
Why do you break the batches?

**Comment by kohjingyu** [comment]
Thanks for pointing this out! I realize this wasn't mentioned in the paper, but we follow the same scheme as [Fromage](https://arxiv.org/abs/2301.13823) in concatenating captions so that the model sees some interleaved image + text data (i.e., `<image><text><image><text>`):
<img width="437" alt="Screen Shot 2023-07-25 at 5 01 18 PM" src="https://github.com/kohjingyu/gill/assets/2196979/76a69db6-7bf0-4316-9a81-463f20f92d14">

We'll add this to the next version of the paper.

**Comment by kohjingyu** [comment]
During the inference for evaluation we do add the [IMG] tokens:

https://github.com/kohjingyu/gill/blob/232eb02812e7e138cd7574772a11d3d8a71d34e9/evals/generate_vist_images.py#L70-L73

If you mean for qualitative inference (e.g., a chatbot-based demo), we don't add it explicitly because we want the model to decide when it should produce [IMG] tokens (rather than always adding images at the end of generated text). Hope that makes sense!

**Comment by Epiphqny** [comment]
@kohjingyu Thanks for your detailed explanation,  could your tell me how to train the generation task only? Have you tested the performance of only training for one generation task? Do I only need to preserve the position and generation prediction loss? Where is the  image token prediction loss in the code? It seems that the losses for generation and retrieval  are tight and I am not sure if it make sense to separate them.

**Comment by kohjingyu** [comment]
If you want to train a generation only model, you can probably do so by changing this line to `model_modes = ['generation']`:

https://github.com/kohjingyu/gill/blob/232eb02812e7e138cd7574772a11d3d8a71d34e9/main.py#L479

I haven't tested this in the latest version and you might need to make some minor changes in other parts of the code.
